src/core_modules/capture-core/src/components/QuickSelector/ProgramSelector.component.js

```diff
@@ -1,63 +1,209 @@
 import { InputLabel } from 'material-ui-next/Input';
 import Paper from 'material-ui-next/Paper';
+import IconButton from 'material-ui-next/IconButton';
+import ClearIcon from 'material-ui-icons/Clear';
+import Grid from 'material-ui-next/Grid';
+
+import { getTranslation } from '../../d2/d2Instance';
 
 const styles = () => ({
     paper: {
         padding: 15,
         backgroundColor: '#f6f6f6',
         borderRadius: 5,
     },
+    title: {
+        margin: 0,
+        fontWeight: 425,
+        fontSize: 15,
+    },
     form: {
         width: '100%',
     },
     listItem: {
         backgroundColor: '#ffffff',
         borderRadius: 5,
-        margin: 5,
+        marginBottom: 5,
+        padding: 5,
+        border: '1px solid lightGrey',
+        '&:hover': {
+            backgroundColor: 'white',
+            borderColor: '#71a4f8', 
+       },
+    },
+    selectedText: {
+        marginTop: 5,
+        marginBottom: 5,
+        marginLeft: 5,
+        width: '100%',
+        padding: 5,
+        borderLeft: '2px solid #71a4f8',
+    },
+    selectedPaper: {
+        backgroundColor: '#f6f6f6',
+        borderRadius: 5,
+        padding: 5,
+    },
+    selectedButton: {
+        float: 'right',
+        width: 20,
+        height: 20,
+    },
+    selectedButtonIcon: {
+        width: 20,
+        height: 20,
+    },
+    programAC: {
+        backgroundColor: 'white',
+        border: '1px solid lightGrey',
+        borderRadius: 5,
+        padding: '0px 5px 2px 5px',
     },
 });
 
 type Props = {
-    handleChangeProgram: (value: string) => void,
     handleClickProgram: (value: string) => void,
+    handleSetCatergoryCombo: (value: string, value: string) => void,
+    resetProgram: () => void,
     selectedProgram: Object,
+    selectedCategories: Object,
     classes: Object,
 };
 
 class ProgramSelector extends Component<Props> {
-    handleChange: (event: any) => void;
     handleClick: (program: Object) => void;
+    handleClickCategoryOption: (value: string, value: string) => void;
     constructor(props) {
         super(props);
-        this.handleChange = this.handleChange.bind(this);
         this.handleClick = this.handleClick.bind(this);
-    }
-
-    handleChange(event) {
-        this.props.handleChangeProgram(event.target.value);
+        this.handleClickCategoryOption = this.handleClickCategoryOption.bind(this);
     }
 
     handleClick(program) {
         this.props.handleClickProgram(program.id);
     }
 
-    // TODO: Add support for cat-combos.
+    handleClickCategoryOption(selectedCategoryOption, categoryId) {
+        this.props.handleSetCatergoryCombo(selectedCategoryOption, categoryId);
+    }
+
+    handleResetProgram() {
+        this.props.resetProgram();
+        
+        // If categoryOptions have been selected, they need to be removed when program is de-selected.
+        if (this.props.selectedCategories) {
+            this.props.handleSetCatergoryCombo(null, null);
```

Comment:
Consider making handleResetCategoryCombo that takes no arguments.

Reply:
Done.